hoodie-client/src/main/java/com/uber/hoodie/HoodieReadClient.java

```diff
@@ -130,7 +124,7 @@ private void assertSqlContext() {
    * @return a dataframe
    */
   public Dataset<Row> read(JavaRDD<HoodieKey> hoodieKeys, int parallelism) throws Exception {
-
+    startEmbeddedServerView(hoodieKeys.map(k -> k.getPartitionPath()).distinct().collect());
```

Comment:
May be a similar method like in HoodieWriteClient (getTableAndInitCtx) and move the table instantiation and embedded server instantiation there and call that in all these methods

Reply:
@bvaradar If we are going to start the timeline in the constructor, does it impact the creation time of that object upfront ?

Reply:
@n3nash : the server startup is pretty quick (200-300 msecs empirically) as filesystem view is loaded lazily